### PR TITLE
Feature: SET_INITIAL_STATE_PTR hint

### PR DESCRIPTION
**Pull request by odesenfans**
Issue Number: N/A

## Type

- [x] feature
- [ ] bugfix
- [ ] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

**Comment by odesenfans** [comment]
Depends on #87.

File: src/starknet/starknet_storage.rs [hunk 1]
```diff
@@ -62,34 +68,127 @@ where
...
+    #[serde_as(as = "Felt252Num")]
+    pub updated_root: Felt252,
+    pub tree_height: usize,
+    #[serde_as(as = "HashMap<Felt252Str, Vec<Felt252Str>>")]
+    pub commitment_facts: HashMap<Felt252, Vec<Felt252>>,
+}
+
+// #[serde_as]
+// #[derive(Debug, Clone, Default, Deserialize, Serialize, PartialEq)]
+// pub struct CommitmentInfo {
```
Comment (notlesh) [inline src/starknet/starknet_storage.rs:86]: Do we need this?

File: src/hints/state.rs [hunk 2]
```diff
@@ -365,4 +447,34 @@ mod tests {
...
+        set_initial_state_updates_ptr(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants)
+            .expect("Hint should succeed");
+
+        let os_input_from_scope: StarknetOsInput = exec_scopes.get(vars::scopes::OS_INPUT).unwrap();
+        assert_eq!(os_input_from_scope, os_input);
+
+        let commitment_info_by_address: HashMap<Felt252, CommitmentInfo> =
+            exec_scopes.get(vars::scopes::COMMITMENT_INFO_BY_ADDRESS).unwrap();
+
+        // TODO: more asserts on the contract commitment info (?)
```
Comment (notlesh) [inline src/hints/state.rs:477]: Is this worth keeping around? Or just doing?

**Review by notlesh — APPROVED** [review]
LGTM, couple nits

File: src/starknet/starknet_storage.rs (hunk 1 above)
Comment (odesenfans) [reply src/starknet/starknet_storage.rs:86]: Nope, removing it.

File: src/hints/state.rs (hunk 2 above)
Comment (odesenfans) [reply src/hints/state.rs:477]: Keeping it for now.

**Pull request merged**